src/rgw/rgw_rados.cc

```diff
@@ -3384,6 +3442,19 @@ class RGWRadosPutObj : public RGWHTTPStreamRWRequest::ReceiveCB
   uint64_t get_data_len() {
     return data_len;
   }
+
+  string get_calculated_etag() {
+    if (!cct->_conf->rgw_sync_obj_etag_verify)
```

Comment:
the value of `rgw_sync_obj_etag_verify` might change at runtime, so we should only check its value once in `RGWRadosPutObj::process_attrs()`. later checks should be based on state that's local to RGWRadosPutObj